### ISSUE TITLE
Filter for previous year, the year before ...

### ISSUE THREAD

**Issue by rnd-nguyenth** [frappe/insights#327]
Hi,

It would be great if there is an option for Filter date such as Previous 1, 2, 3 ... year
![image](https://github.com/user-attachments/assets/9e668957-7a9e-4471-a32b-41b2b1f6c903)

**Comment by nextchamp-saqib** [comment]
You can already apply a filter on date fields to show data from "previous 1 year"
I didn't understand your case, can you please explain with an example?

**Comment by rnd-nguyenth** [comment]
Hi Saqib, for example, I had data from 2022 to 2024, I filtered **Last 1 Year** and it showed data for 2023 which is correct. Then I filtered **Last 2 Year**, it showed both 2022 and 2023. What I want to achieve is it is only showing for 2022. I guess I can select between and choose from 2022-01-01 to 2022-12-31.

**Comment by JeansReal** [comment]
Hello is there any filter to show from today to last year?

For example December 2024(Currrent Month) to December 2023(last 1 year, 12 months)

**Comment by nextchamp-saqib** [comment]
> Hi Saqib, for example, I had data from 2022 to 2024, I filtered Last 1 Year and it showed data for 2023 which is correct. Then I filtered Last 2 Year, it showed both 2022 and 2023. What I want to achieve is it is only showing for 2022. I guess I can select between and choose from 2022-01-01 to 2022-12-31.

Yes, the easiest way would be to set a `between` filter, or if you want to make it dynamic, I believe, it can be done using a custom expression. 

> Hello is there any filter to show from today to last year? For example December 2024(Currrent Month) to December 2023(last 1 year, 12 months)

You can add two filters: `date within last 1 year` or `date within current year` (only possible in v3)
